src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,67 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './FindMovie.scss';
+import cn from 'classnames';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+import { getMovie } from '../../api';
+
+interface Props {
+  addMovie: (movie: Movie) => void;
+}
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [error, setError] = useState(false);
+  const [isLoading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setError(false);
+  }, [query]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuery(value);
+  };
+
+  const handleMovieAdd = () => {
+    if (movie) {
+      addMovie(movie);
+      setMovie(null);
+      setQuery('');
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setLoading(true);
+
+    getMovie(query)
+      .then(result => {
+        if ('Error' in result) {
+          setError(true);
+        } else {
```

Comment:
can you do it without `if / else` construction?

Reply:
not really sure how to do it since `catch` not working cuz 'error' is legit response

Reply:
just make `return` if you have an error